crates/biome_migrate/src/analyzers/nursery_rules.rs

```diff
@@ -0,0 +1,299 @@
+use crate::{declare_migration, MigrationAction};
+use biome_analyze::context::RuleContext;
+use biome_analyze::{ActionCategory, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_diagnostics::{category, Applicability};
+use biome_json_factory::make::{
+    json_member, json_member_list, json_member_name, json_object_value, json_string_literal, token,
+};
+use biome_json_syntax::{AnyJsonValue, JsonMember, JsonMemberList, JsonRoot, T};
+use biome_rowan::{
+    AstNode, AstNodeExt, AstSeparatedList, BatchMutationExt, TextRange, TriviaPieceKind, WalkEvent,
+};
+use std::collections::HashMap;
+
+#[cfg(not(debug_assertions))]
+use crate::analyzers::rules_to_migrate::RULES_TO_MIGRATE;
+
+declare_migration! {
+    pub(crate) NurseryRules {
+        version: "1.7.0",
+        name: "nurseryRules",
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct MigrateRuleState {
+    /// The member of the new rule
+    nursery_rule_member: JsonMember,
+    /// The member of the group where the new rule should be moved
+    nursery_group: JsonMember,
+    /// The name of the new rule
+    new_rule_name: &'static str,
+    /// The new group name
+    new_group_name: &'static str,
+}
+
+impl MigrateRuleState {
+    fn as_rule_name_range(&self) -> TextRange {
+        self.nursery_rule_member.range()
+    }
+}
+
+/// It attempts the find the group by name:
+///
+/// ```json5
+/// {
+///     "groupName": {}
+/// //  ^^^^^^^^^^^
+/// }
+/// ```
+fn find_group_by_name(root: &JsonRoot, group_name: &str) -> Option<JsonMember> {
+    let preorder = root.syntax().preorder();
+    let mut group = None;
+    for event in preorder {
+        if let WalkEvent::Enter(node) = event {
+            let Some(member) = JsonMember::cast(node) else {
+                continue;
+            };
+            let Ok(text) = member.name().and_then(|n| n.inner_string_text()) else {
+                continue;
+            };
+            if text.text() == group_name {
+                group = Some(member);
+                break;
+            }
+        }
+    }
+
+    group
+}
+
+// used for testing purposes
+/// - Left: name of the rule in the nursery group
+/// - Right: name of the new group and name of the new rule (sometimes we change name)
+#[cfg(debug_assertions)]
+const RULES_TO_MIGRATE: [(&str, (&str, &str)); 2] =
```

Comment:
This array is expected to change often. We could add an indirection to make its edition simpler.
```suggestion
const RULES_TO_MIGRATE: &[(&str, (&str, &str)); 2] = &
```

Reply:
Can you remove the suggestion? This change won't work with the current code (I know because I already tried it)